unittests/NodeAPI/CMakeLists.txt

```
@@ -75,15 +75,27 @@ add_subdirectory(js)
 #   $<TARGET_FILE:libshared>
 #   $<TARGET_FILE_DIR:NodeApiTests>)
 
-add_executable(node_lite
-  child_process.cpp
-  child_process.h
-  node_lite.cpp
-  node_lite.h
-  node_lite_hermes.cpp
-  string_utils.cpp
-  string_utils.h
-)
+if(APPLE)
+  add_executable(node_lite
+    child_process_mac.cpp
+    child_process.h
+    node_lite.cpp
+    node_lite.h
+    node_lite_hermes.cpp
+    string_utils.cpp
+    string_utils.h
+  )
+else()
+  add_executable(node_lite
+    child_process.cpp
+    child_process.h
+    node_lite.cpp
+    node_lite.h
+    node_lite_hermes.cpp
+    string_utils.cpp
+    string_utils.h
+  )
+endif()
```

To avoid repeating the source files which are not platform dependent, you could use the `set` and `list(APPEND`

```suggestion
set(NODE_LITE_SOURCES
    child_process.h
    node_lite.cpp
    node_lite.h
    node_lite_hermes.cpp
    string_utils.cpp
    string_utils.h
)

if(APPLE)
  list(APPEND NODE_LITE_SOURCES child_process_mac.cpp)
else()
  list(APPEND NODE_LITE_SOURCES child_process.cpp)
endif()

add_executable(node_lite ${NODE_LITE_SOURCES})
```

Or use `target_sources` to build up the sources for the target, using multiple (conditional) invocations - this is a bit simpler as it gets rid of the extra `NODE_LITE_SOURCES` variable:

```
add_executable(node_lite)

target_sources(node_lite
  PRIVATE
    child_process.h
    node_lite.cpp
    node_lite.h
    node_lite_hermes.cpp
    string_utils.cpp
    string_utils.h
)

if(APPLE)
  target_sources(node_lite PRIVATE child_process_mac.cpp)
else()
  target_sources(node_lite PRIVATE child_process.cpp)
endif()
```

---

Resolved in `reduce redundancy in list of sources`.

Thanks!